### DISCUSSION TITLE
Performance issues -  feature insert via feature dao an content values

### DISCUSSION THREAD

**Discussion by pab2d** [ngageoint/geopackage-android#73] [General]
Hi,

using GeoPackage, i recognizes that the insert perfomance is far slower than my privious SpatiaLite implementation with a bulk insert.

privious:
my BulkInsert class creates a sql string with an insert command and all content values..
```
private BulkInsert(String tableName, ContentValues cv) throws Exception {
	this.db = getDBReadWrite();
	this.db.exec("BEGIN TRANSACTION", null);
	this.stmt = prepareInsertStatement(db, tableName, cv, true);
}
```
Inside the prepareInsertStatement methode the function: ST_GeomFromWBK is used to handle the geometry column

```
BulkInsert bi=null;
...
long id=bi.execute(cv);
```

current (GeoPackage):
```
FeatureDao featureDao = gpkg.getFeatureDao(tableName);
rowCount = featureDao.insert(cv);
```

Is there a way to increase the insert performance?

**Comment by bosborn** [comment]
Try performing inserts in chunked transactions.

https://github.com/ngageoint/geopackage-android/blob/3679d011ab1e8f05810f209c97d61cf757945861/geopackage-sdk/src/main/java/mil/nga/geopackage/user/UserDao.java#L116-L146


See Example Tests:
* [Performance Test on Feature Inserts](https://github.com/ngageoint/geopackage-android/blob/3679d011ab1e8f05810f209c97d61cf757945861/geopackage-sdk/src/androidTest/java/mil/nga/geopackage/GeoPackagePerformance.java#L128)
* [Transaction Tests](https://github.com/ngageoint/geopackage-android/blob/3679d011ab1e8f05810f209c97d61cf757945861/geopackage-sdk/src/androidTest/java/mil/nga/geopackage/db/TransactionTest.java#L30)

**Reply by pab2d** [reply]
Thank you bosborn
